web/server/codechecker_server/session_manager.py

```diff
@@ -650,6 +681,60 @@ def create_session(self, auth_string):
 
         return local_session
 
+    def create_session_oauth(self, provider, username, token):
+        """ Creates a new session for the given auth-string. """
+
+        if not self.__is_method_enabled('oauth'):
+            return False
+
+        # Try to get the user's previous session.
+        for sess in self.__sessions:
+            if sess.user == username:
+                return sess
```

Comment:
So if you log in from multiple places into the same account, all browsers will have the same session? That doesn't seem like a great idea - what if you log out from any of the browsers? What if your session cookie gets stolen? What if it gets refreshed in one browser, but not in others?

In general, where you want to keep state is associating the CodeChecker session cookie with the OAuth session/refresh tokens. Every browser should have a separate, independent session.